Add option to generate tests.

srmurguia opened the pull request:
Now tests are also created if generateTests is set to true.

srmurguia committed:
Add option to generate tests.

Now tests are also created if generateTests is set to true.

NikitaKozlov commented on src/main/kotlin/com/google/androidstudiopoet/generators/packages/KotlinGenerator.kt:
```diff
@@ -44,11 +44,38 @@ class KotlinGenerator constructor(fileWriter: FileWriter) : PackageGenerator(fil
+
+        buff.append("import org.junit.Test\n\n")
+
+        buff.append("public class ${blueprint.className + "Test"} {\n")
```
To avoid duplication between java and kotlin generators, please move `blueprint.className + "Test"` to the `ClassBlueprint`. Same for "MethodBlueprint" and test name.

srmurguia committed:
Move test clasess to its own ClassBlueprint.

Doing so improves code reuse and makes test code generation cleaner.
Also, made a couple of changes to generated code alignment.
